### PR TITLE
J-292: + added show password icon on the auth pages

### PR DESCRIPTION
**Pull request by JuliaDzyuba**

### CODE REVIEW

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx [hunk 1]
```diff
@@ -67,14 +74,21 @@ const SignUpForm: React.FC<Props> = ({ onSubmit }) => {
...
+            placeholder="Enter your password"
+            name={UserPayloadKey.PASSWORD}
+            control={control}
+            errors={errors}
+          />
+          <button
+            type="button"
+            className={isVisible ? getAllowedClasses(styles.showPasswordBtn, styles.visible) : styles.showPasswordBtn}
+            onClick={handleToggleVisible}
+          ></button>
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]:
```suggestion
          >
            <span className="visually-hidden">Toggle password visibility</span>
          </button>
```

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 2]
```diff
@@ -47,14 +54,21 @@ const SignInForm: React.FC<Props> = ({ onSubmit }) => {
...
+            placeholder="Enter your password"
+            name={UserPayloadKey.PASSWORD}
+            control={control}
+            errors={errors}
+          />
+          <button
+            type="button"
+            className={isVisible ? getAllowedClasses(styles.showPasswordBtn, styles.visible) : styles.showPasswordBtn}
+            onClick={handleToggleVisible}
+          ></button>
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx]:
```suggestion
          >
            <span className="visually-hidden">Toggle password visibility</span>
          </button>
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/assets/img/eye-invisible.svg [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (Denidzio) [inline frontend/src/assets/img/eye-invisible.svg]: Let's [compress ](shrinkme.app) it
`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 469.44 469.44">
    <path d="M231.147 160.373l67.2 67.2.32-3.52c0-35.307-28.693-64-64-64l-3.52.32z"/>
    <path d="M234.667 117.387c58.88 0 106.667 47.787 106.667 106.667 0 13.76-2.773 26.88-7.573 38.933l62.4 62.4c32.213-26.88 57.6-61.653 73.28-101.333-37.013-93.653-128-160-234.773-160-29.867 0-58.453 5.333-85.013 14.933l46.08 45.973c12.052-4.693 25.172-7.573 38.932-7.573zM21.333 59.253l48.64 48.64 9.707 9.707C44.48 145.12 16.64 181.707 0 224.053c36.907 93.653 128 160 234.667 160 33.067 0 64.64-6.4 93.547-18.027l9.067 9.067 62.187 62.293 27.2-27.093L48.533 32.053l-27.2 27.2zM139.307 177.12l32.96 32.96c-.96 4.587-1.6 9.173-1.6 13.973 0 35.307 28.693 64 64 64 4.8 0 9.387-.64 13.867-1.6l32.96 32.96c-14.187 7.04-29.973 11.307-46.827 11.307-58.88 0-106.667-47.787-106.667-106.667 0-16.853 4.267-32.64 11.307-46.933z"/>
</svg>`

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 4]
```diff
@@ -47,14 +54,21 @@ const SignInForm: React.FC<Props> = ({ onSubmit }) => {
...
+            type={isVisible ? InputType.TEXT : InputType.PASSWORD}
+            label="Password"
+            placeholder="Enter your password"
+            name={UserPayloadKey.PASSWORD}
+            control={control}
+            errors={errors}
+          />
+          <button
+            type="button"
+            className={isVisible ? getAllowedClasses(styles.showPasswordBtn, styles.visible) : styles.showPasswordBtn}
```
Comment (Denidzio) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx]:
```suggestion
            className={getAllowedClasses(styles.showPasswordBtn, isVisible && styles.visible)}
```

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx [hunk 5]
```diff
@@ -7,10 +7,11 @@ import {
   UserPayloadKey,
 } from 'common/enums/enums';
 import { signUp as signUpValidationSchema } from 'validation-schemas/validation-schemas';
-import { useAppForm, useAppSelector } from 'hooks/hooks';
+import { useAppForm, useAppSelector, useState } from 'hooks/hooks';
 import { Button, Input, Link, Datepicker } from 'components/common/common';
 import { DEFAULT_REGISTER_PAYLOAD } from './common/constants';
 import styles from './styles.module.scss';
+import { getAllowedClasses } from 'helpers/helpers';
```
Comment (Denidzio) [inline frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]: Import order

File: frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx [hunk 6]
```diff
@@ -67,14 +74,21 @@ const SignUpForm: React.FC<Props> = ({ onSubmit }) => {
...
+            type={isVisible ? InputType.TEXT : InputType.PASSWORD}
+            label="Password"
+            placeholder="Enter your password"
+            name={UserPayloadKey.PASSWORD}
+            control={control}
+            errors={errors}
+          />
+          <button
+            type="button"
+            className={isVisible ? getAllowedClasses(styles.showPasswordBtn, styles.visible) : styles.showPasswordBtn}
```
Comment (Denidzio) [inline frontend/src/components/auth/components/sign-up-form/sign-up-form.tsx]:
```suggestion
           className={getAllowedClasses(styles.showPasswordBtn, isVisible && styles.visible)}
```

**Review by Denidzio — CHANGES_REQUESTED** [review]

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 7]
```diff
@@ -28,6 +29,12 @@ const SignInForm: React.FC<Props> = ({ onSubmit }) => {
 
   const isFormDisable = authStatus === DataStatus.PENDING;
 
+  const [isVisible, setIsVisible] = useState<boolean>(false);
```
Comment (Denidzio) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx]: It is not entirely clear from the name what is visible

File: frontend/src/assets/img/eye-invisible.svg [hunk 8]
```diff
@@ -1,50 +1 @@
...
-<g>
-</g>
-<g>
-</g>
-<g>
-</g>
-<g>
-</g>
-</svg>
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 469.44 469.44"> <path d="M231.147 160.373l67.2 67.2.32-3.52c0-35.307-28.693-64-64-64l-3.52.32z"/> <path d="M234.667 117.387c58.88 0 106.667 47.787 106.667 106.667 0 13.76-2.773 26.88-7.573 38.933l62.4 62.4c32.213-26.88 57.6-61.653 73.28-101.333-37.013-93.653-128-160-234.773-160-29.867 0-58.453 5.333-85.013 14.933l46.08 45.973c12.052-4.693 25.172-7.573 38.932-7.573zM21.333 59.253l48.64 48.64 9.707 9.707C44.48 145.12 16.64 181.707 0 224.053c36.907 93.653 128 160 234.667 160 33.067 0 64.64-6.4 93.547-18.027l9.067 9.067 62.187 62.293 27.2-27.093L48.533 32.053l-27.2 27.2zM139.307 177.12l32.96 32.96c-.96 4.587-1.6 9.173-1.6 13.973 0 35.307 28.693 64 64 64 4.8 0 9.387-.64 13.867-1.6l32.96 32.96c-14.187 7.04-29.973 11.307-46.827 11.307-58.88 0-106.667-47.787-106.667-106.667 0-16.853 4.267-32.64 11.307-46.933z"/> </svg>
```
Comment (Denidzio) [inline frontend/src/assets/img/eye-invisible.svg:1]: You can format this. Nothing bad will happen

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 9]
```diff
@@ -29,10 +29,10 @@ const SignInForm: React.FC<Props> = ({ onSubmit }) => {
 
   const isFormDisable = authStatus === DataStatus.PENDING;
 
-  const [isVisible, setIsVisible] = useState<boolean>(false);
+  const [isVisiblePassword, setIsVisiblePassword] = useState<boolean>(false);
 
   const handleToggleVisible = (): void => {
```
Comment (Denidzio) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx:34]: `handleToggleVisible` what? Let's rename it too

**Review by Denidzio — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**